### ISSUE TITLE
Support BoundDynamicObjectInitializerMember in IOperation

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#27499]
```C#
#pragma warning disable 0169
class A
{
    dynamic this[int x, int y]
    {
        get
        {
            return new A();
        }
    }

    dynamic this[string x, string y]
    {
        get
        {
            throw null;
        }
    }

    int X, Y, Z;

    static void Main()
    {
        dynamic x = 1;
        new A {[y: x, x: x] = { X = 1, Y = 1, Z = 1 } };
    }
}
```

If you get the IOperation tree for the object creation, you'll see OperationKind.None for the A.X, A.Y, and A.Z references in the initializer.

**Comment by jinujoseph** [comment]
@333fred  has this been addressed already ?

**Comment by 333fred** [comment]
Yes, this was fixed in #27519.
